### PR TITLE
Add package_data to setup.py for hdaccount wordlists

### LINKED ISSUE #102: Mnemonics are badly broken
## What was wrong?

Mnemonics only work on a from-source install of eth-account. Released versions are broken with:

```py
>>> from eth_account import Account
>>> Account.enable_unaudited_hdwallet_features()
>>> Account.create_with_mnemonic()
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/jcarver/code/eth-account/tmp/non-develop-venv/lib/python3.6/site-packages/eth_utils/decorators.py", line 20, in _wrapper
    return self.method(objtype, *args, **kwargs)
  File "/home/jcarver/code/eth-account/tmp/non-develop-venv/lib/python3.6/site-packages/eth_account/account.py", line 333, in create_with_mnemonic
    mnemonic = generate_mnemonic(num_words, language)
  File "/home/jcarver/code/eth-account/tmp/non-develop-venv/lib/python3.6/site-packages/eth_account/hdaccount/__init__.py", line 16, in generate_mnemonic
    return Mnemonic(lang).generate(num_words)
  File "/home/jcarver/code/eth-account/tmp/non-develop-venv/lib/python3.6/site-packages/eth_account/hdaccount/mnemonic.py", line 79, in __init__
    f'Invalid language choice "{language}", must be one of {languages}'
eth_utils.exceptions.ValidationError: Invalid language choice "english", must be one of []
```

## How can it be fixed?

1. Make sure that tests run against a "clean-room" installed version of eth-account (at least for the mnemonic tests)
2. Package up the language files to include in the release

### PR DESCRIPTION
**Pull request by kclowes**
## What was wrong?
Web3py couldn't use the new HD Account methods because the wordlists (raw txt files) weren't getting included in the release.

Fixes #102 

## How was it fixed?
Added a line to setup.py to include them, and tested in web3 using a local install of eth-account.

#### Cute Animal Picture

image

### CODE REVIEW

**Comment by pipermerriam** [comment]
I *think* CI might have caught this were we to have `usedevelop=False` in the `tox.ini` config.

https://github.com/ethereum/eth-account/blob/master/tox.ini#L24

I'm not 100% sure about this but I do believe this is something that can and should be caught by CI.

**Comment by kclowes** [comment]
Yeah, agreed. It would have been nice to catch this in CI. I'm getting a bunch of `ImportMismatchError`s when I remove `usedevelop=true` and my google-fu is failing me. Do you know a way around that?

**Comment by kclowes** [comment]
Ha, never mind. Fixed it! Let me know if there is a better way to fix it!

File: setup.py [hunk 1]
```diff
@@ -47,6 +47,18 @@
     author_email='snakecharmers@ethereum.org',
     url='https://github.com/ethereum/eth-account',
     include_package_data=True,
+    package_data={"eth_account": ["hdaccount/wordlist/*.txt"]},
+    data_files=[('eth_account', [
+        'eth_account/hdaccount/wordlist/chinese_simplified.txt',
```
Comment (pipermerriam) [inline setup.py:52]: I think you can do this via the `MANIFEST.in` file which allows for a `*` style inclusion that will not suffer from us forgetting to update this list if we add a new word-list.

https://github.com/ethpm/ethpm-cli/blob/432e3aa40faeda6eea54530bdf3d2c81fd80d5a1/MANIFEST.in#L41

Comment (kclowes) [reply setup.py:52]: We have a `Manifest.in`, but the doctests were still failing. I _think_ we only need this for the doctests, in which case we only need the english one, so I could just include that one?

**Comment by kclowes** [comment]
That makes sense, but that was the only way I could get around the `ImportMismatchError` after I changed `usedevelop` to `false`. Do you know another way to get around that error?

**Comment by kclowes** [comment]
I also haven't figured out why quite yet, but setting `usedevelop=false` doesn't fail any tests if the `wordlist/*.txt` files aren't packaged up, even though from everything I've read, it seems like it should. The doctests do fail as expected without the wordlist files, however. Maybe there is something special about what gets installed for tests in the `tests` dir?

**Comment by carver** [comment]
Yeah, so the way I've usually been testing this kind of thing is a test where it specifically builds a wheel and then only uses that for running the test. It's a little ugly, but it's a way to know that exactly the installed version is being tested. Like:

https://github.com/ethereum/trinity/blob/27bbf61209d81b894292e36904a8b634fcb0094a/tox.ini#L101-L105

**Comment by carver** [comment]
Notice the `skip_install` in addition to not using develop:
https://github.com/ethereum/trinity/blob/27bbf61209d81b894292e36904a8b634fcb0094a/tox.ini#L111-L112

File: setup.py (hunk 1 above)
Comment (kclowes) [reply setup.py:52]: @carver @pipermerriam So turns out if I use the generated tarball file to install `eth-account`, I need the `data_files` here, but if I use the wheel to install, I don't need data_files. How concerned are we about the tarball distribution? I assume both should work?

Comment (carver) [reply setup.py:60]: Hm, this suggests that the wheel is better, all else equal: https://stackoverflow.com/a/42226284/8412986

Based on the docs though, I can't see why we would need `data_files` *and* `package_data`. So you're saying a wheel install will fail when this `data_files` entry is removed?

Comment (kclowes) [reply setup.py:60]: Yeah, it's entirely possible I'm doing something wrong, but when I do 
```
make dist # in my eth-account dir, 
cd path/to/web3 
pip install /path/to/eth-account/eth_account/dist/eth-account-0.5.0.tar.gz
```
and open up ipython to test, I get the error where it doesn't load any of the languages from eth-account's `Mnemonic.list_languages` (code is found [here](https://github.com/ethereum/eth-account/blob/master/eth_account/hdaccount/mnemonic.py#L85-L86)). The `make dist` output shows that the files get added. My one thought was that maybe there is something going on with relative/absolute paths in the `list_languages` method?

Comment (kclowes) [reply setup.py]: Got it! The syntax was wrong in the `MANIFEST.in` file

File: MANIFEST.in [hunk 2]
```diff
@@ -5,7 +5,7 @@ include requirements-docs.txt
 global-include *.pyi
 
 # For mnemonics
-recursive-include eth_account/hdaccount/wordlist/*
+recursive-include eth_account hdaccount wordlist *.txt
```
Comment (carver) [inline MANIFEST.in:8]: Aha good find! By my reading of [this](https://packaging.python.org/guides/using-manifest-in/):

> recursive-include dir-pattern pat1 pat2 ... | Include all files under directories matching dir-pattern that match any of the listed patterns

I would expect that the intended approach is:
```suggestion
recursive-include eth_account/hdaccount/wordlist *.txt
```

Does that work?

Comment (carver) [reply MANIFEST.in:8]: Also, it's worth making sure you have the latest tooling with a `pip install -U pip setuptools wheel`. They have seemed pretty active.

**Comment by carver** [comment]
It bothers me that CI was still green before the manifest syntax. Would it be possible to add the wheel-building test in, show that it fails, then push the last manifest syntax fix to show that turns it green?

**Comment by kclowes** [comment]
@carver I think this is ready for a last look. I added the wheel-cli tests to circleci. Let me know if I should test for anything else there. Since I removed the `usedevelop=True`, the mnemonics core tests are failing in the way that I'd expect, so I didn't add that to the wheel-cli tests. Since eth-account doesn't really have cli tests like trinity, I made the tests more closely resemble the web3 tests where I just made sure that eth-account could be imported after install. I did check locally that it worked, but can push up a test if you're interested in seeing it go red.

**Pull request merged**